### ISSUE TITLE
[feature-request] Show code in diagnostic message

### ISSUE THREAD

**Issue by smackesey** [charliermarsh/ruff-lsp#9]
Notice the absence of the code in the diagnostic text here, makes it hard to know how to ignore the error.

<img width="654" alt="image" src="https://user-images.githubusercontent.com/1531373/209043416-85c36f50-aa65-4bc0-bdbe-2f949726a879.png">

**Comment by charliermarsh** [comment]
Hmm, it might be implementation-dependent, it does show the diagnostic code in VS Code and in Sublime. I'll look into it.

![Screen Shot 2022-12-21 at 10 17 16 PM](https://user-images.githubusercontent.com/1309177/209048106-af6158ce-19b0-485e-8d7a-a4c86c3b71db.png)

**Comment by smackesey** [comment]
@WhyNotHugo It is indeed neovim, thanks for the pointer. Apparently you need to use a custom diagnostic format function to show the code. Was able to fix with this config snippet:

```
-- Required to get codes to show up.
local function fmt(diagnostic)
  if diagnostic.code then
    return ("[%s] %s"):format(diagnostic.code, diagnostic.message)
  end
  return diagnostic.message
end

-- Always show the diagnostic source in its text representation in the
-- floating window.
vim.diagnostic.config({
  float = {
    -- Always show the diagnostic source in its text representation in the
    -- floating window.
    format = fmt,
    source = 'always',
  }
})
```
